File: bender/app/src/main/java/kz/app/bender/mvp/tabsNavigation/FlowFragmentMainPresenter.java
package kz.app.bender.mvp.tabsNavigation;

import com.arellomobile.mvp.InjectViewState;

import kz.app.bender.mvp.AbstractPresenter;
import kz.app.bender.support.Config;
import kz.app.bender.support.navigation.BenderRouter;
import kz.app.bender.support.navigation.Screens;
import kz.app.bender.support.navigation.screen.BenderScreen;

@InjectViewState
public class FlowFragmentMainPresenter
        extends AbstractPresenter<FlowFragmentContract.View>
        implements FlowFragmentContract.Presenter {

    //region Vars
    private BenderRouter tabsRouter;
    private BenderScreen screenMain;
    private BenderScreen screenWebSiteTab2;
    private BenderScreen screenWebSiteTab3;
    private FlowFragmentModel flowFragmentModel = new FlowFragmentModel();
    //endregion

    //region Constructor
    FlowFragmentMainPresenter(BenderRouter flowRouter, BenderRouter tabsRouter) {
        super(flowRouter);
        this.tabsRouter = tabsRouter;
    }
    //endregion

    //region Overridden methods
    @Override
    protected void onFirstViewAttach() {
        super.onFirstViewAttach();
        onTabClicked(Screens.Tabs.tab_login);
    }

    @Override
    public void onDestroy() {
        screenMain = null;
        screenWebSiteTab2 = null;
        screenWebSiteTab3 = null;
        super.onDestroy();
    }
    //endregion

    //region Actions
    void onTabClicked(Screens.Tabs tab) {
        BenderScreen screen = null;
        if (tab == Screens.Tabs.tab_login) {
            if (screenMain == null) {
                if (flowFragmentModel.getUserData() != null)
                    screenMain = Screens.getTabChildrenList();
                else
                    screenMain = Screens.getTabLogin();
            }
            screen = screenMain;
        } else if (tab == Screens.Tabs.tab_web_site_karlygash) {
            if (screenWebSiteTab2 == null)
                screenWebSiteTab2 = Screens.getTabWebView(Config.URL_SITE_SIMPLE);
            screen = screenWebSiteTab2;
        } else if (tab == Screens.Tabs.tab_web_site_profile) {
            if (screenWebSiteTab3 == null)
                screenWebSiteTab3 = Screens.getTabWebView2(Config.URL_SITE_CABINET);
            screen = screenWebSiteTab3;
        }

        if (screen != null)
            tabsRouter.navigateTo(screen);
    }
    //endregion

    //region Support
    void setTabsRouter(BenderRouter router) {
        this.tabsRouter = router;
    }
    //endregion
}

File: DeliveryApp/app/src/main/java/kz/app/deliveryapp/mvp/tabsNavigation/FlowFragmentMainPresenter.kt
package kz.app.deliveryapp.mvp.tabsNavigation

import com.arellomobile.mvp.InjectViewState
import kz.app.deliveryapp.mvp.AbstractPresenter
import kz.app.deliveryapp.support.навигация.BenderRouter
import kz.app.deliveryapp.support.навигация.Screens
import kz.app.deliveryapp.support.навигация.screen.BenderScreen

@InjectViewState
class FlowFragmentMainPresenter(
    flowRouter: BenderRouter, //region Vars
    private var tabsRouter: BenderRouter?
) : AbstractPresenter<FlowFragmentContract.View>(flowRouter), FlowFragmentContract.Presenter {

    private var screenMenu: BenderScreen? = null
    private var screenFavorites: BenderScreen? = null
    private var screenNews: BenderScreen? = null
    private var screenAksii: BenderScreen? = null

    companion object {
        lateinit var instance: FlowFragmentMainPresenter
    }

    init {
        instance = this
    }

    override fun onFirstViewAttach() {
        super.onFirstViewAttach()
        onNavTabClicked(Screens.Tabs.nav_tab_menu)
    }

    override fun onDestroy() {
        screenMenu = null
        screenFavorites = null
        screenNews = null
        screenAksii = null
        super.onDestroy()
    }

    fun onNavTabClicked(tab: Screens.Tabs) {
        var screen: BenderScreen? = null
        when (tab) {
            Screens.Tabs.nav_tab_menu -> {
                if (screenMenu == null)
                    screenMenu = Screens.getNavTabMenu()
                screen = screenMenu
            }
            Screens.Tabs.nav_tab_favorites -> {
                if (screenFavorites == null)
                    screenFavorites = Screens.getNavTabMenu()
                screen = screenFavorites
            }
            Screens.Tabs.nav_tab_news -> {
                if (screenNews == null)
                    screenNews = Screens.getNavTabNews()
                screen = screenNews
            }
            Screens.Tabs.nav_tab_aksii -> {
                if (screenAksii == null)
                    screenAksii = Screens.getNavTabAksii()
                screen = screenAksii
            }
        }

        if (screen != null)
            tabsRouter!!.navigateTo(screen)
    }

    fun setTabsRouter(router: BenderRouter) {
        this.tabsRouter = router
    }
}
